### ISSUE TITLE
Issues with spelling after switching to Vocabularies

### ISSUE THREAD

**Issue by wildmanonline** [jdkato/txtlint#454]
I was looking to identify the cause and implement a fix to a separate issue (the possessive form `'s` of words in the ignore list started resulting in errors at some point over the last few months), when I encountered this issue I'm reporting now. I switched to using Vocabularies instead of using a custom spelling rule with an ignore list. To do this, I did the following:

- Added `Vocab = Custom` in *.vale.ini*
- Added `BasedOnStyles = Vale, CustomStyle` in *.vale.ini* (was previously `BasedOnStyles = CustomStyle`)
- Removed custom `styles > CustomStyle > Spelling.yml` rule file, which extended spelling and ignored a list of words from a `Styles > Dictionary.txt` file.
- Created `styles > Vocab > Custom > accept.txt` and added contents of `styles > Dictionary.txt`file, then deleted the original file.

After doing this, not all spelling errors are reported in Vale. For instance, prior to the change (and after reverting back), the following words would all be reported as `errors`. When I switch to using Vale vocabularies, only `independant` and `maintainance` are marked as errors.

```
eror, noticable, accidentaly, independant, maintainance, preceed, occassionally
```

That's only the tip of the issues with spelling. Single letters, random groups of letters, words ending in `ror`, all of these reported as errors correctly before the switch, and are seemingly ignored and left unreported when using vocabularies.

I'm testing this on macOS 12.4 using Vale CLI v2.18.0 (also tested using v2.19.0 with same result) and the Visual Studio Code extension.

Any ideas on where to go from here? FWIW, it seems like my original issue (possessives forms of ignored/accepted words) is fixed when using vocabularies. But it all seems very wonky so I'm not 100% on that.

**Comment by jdkato** [comment]
> ... the possessive form 's of words in the ignore list started resulting in errors at some point over the last few months ...

If you want the exact behavior you had before, change your `spelling` rule to:

```yaml
extends: spelling
message: "Did you really mean '%s'?"
level: error
filters:
  # You could make this more specific using something
  # like `\w+'s`, but this is what is was before.
  - '\W'
ignore: ../dictionary.txt
```

I don't think this is actually desirable behavior, though, which is why it was changed in [v2.18.0](https://github.com/errata-ai/vale/releases/tag/v2.18.0). If you find that you have a lot of same-word variations you need to cover, I'd suggest providing an actual Hunspell-compatible dictionary ([as shown here](https://vale.sh/docs/topics/styles/#spelling)) instead of using regex.

> ... After doing this, not all spelling errors are reported in Vale.

Entries in [Vocabularies](https://vale.sh/docs/topics/vocab/) are treated as regular expressions, not simply words. And assuming [this](https://github.com/wildmanonline/docs/blob/develop/ci/vale/dictionary.txt) is the `dictionary.txt` file you're referencing, lines like [`le`](https://github.com/wildmanonline/docs/blob/1c8d0edf051f686a8add3a796729284ce91e2796/ci/vale/dictionary.txt#L1036) are going to match a lot of words.

Personally, I don't think something like `le` belongs in an `accept.txt` file. However, if you want to keep it and use Vocabularies, you'll need to make it more specific (like `\ble\b`).

**Comment by wildmanonline** [comment]
Thanks @jdkato! Appreciate the help and fast response. Off the top of your head, do have any good resources for creating Hunspell-compatible dictionaries?

**Comment by jdkato** [comment]
It's not that well documented anywhere, unfortunately. You can check out Vale's [built-in dictionary](https://github.com/errata-ai/en_US-web) for an example or use the [man pages](https://github.com/hunspell/hunspell#documentation).

If you don't want to have to learn the dictionary syntax, just cleaning up your vocabulary entries will probably get you pretty far.
